### PR TITLE
Feat(Middleware): Add Middleware Functionality

### PR DESCRIPTION
**Pull request by toksdotdev**

### CODE REVIEW

File: test/src/require.js [hunk 1]
```diff
@@ -71,6 +89,32 @@ test('testing', (t) => {
   // function to call
   t.is('function', typeof (router.stack[3].route.stack[0].handle));
 
+  // MIDDLEWARES
+  t.is(undefined, router.stack[4].route.stack[1].path);
```
Comment (aichbauer) [inline test/src/require.js]: what is this exactly? why do we test if something is undefined? is it not undefined if we use no middleware?

File: src/index.js [hunk 2]
```diff
@@ -46,7 +69,7 @@ const mapRoutes = (routes, pathToController) => {
       contr = new handler();
     }
 
-    router.route(myPath)[requestMethod](contr[controllerMethod]);
+    router.route(myPath)[requestMethod](middlewares, contr[controllerMethod]);
```
Comment (aichbauer) [inline src/index.js:82]: I should dive more into the Router setup, but I guess if the middleware is an empty array the route is  created without middleware and directly calls the controller method, right?

File: package.json [hunk 3]
```diff
@@ -67,4 +67,4 @@
     "sourceMap": false,
     "instrument": false
   }
-}
+}
```
Comment (aichbauer) [inline package.json]: could you add a newline at the end of this file?

File: README.md [hunk 4]
```diff
@@ -171,3 +201,16 @@ const SomeController = () => {
...
+
+1. Fork it!
+2. Create your feature branch: `git checkout -b feature-name`
+3. Commit your changes: `git commit -am 'Some commit message'`
+4. Push to the branch: `git push origin feature-name`
+5. Submit a pull request 😉😉
+
+
+## License
+This project is licensed under the _____ License - see the [LICENSE.md](LICENSE) file for details.
```
Comment (aichbauer) [inline README.md]: ____ should be MIT.

File: README.md [hunk 5]
```diff
@@ -74,8 +75,40 @@ const UserController = () => {
...
+const routes = {
+  'GET /user:id': {
+    path: 'UserController.get',
+    middlewares: [
+         checkIfAutheticated,
+         verifyFacebookAuth,
+    ],
+    
+  'POST /user': 'UserController.create'
+  },
```
Comment (aichbauer) [inline README.md:124]: This `}` is on the wrong location... `'POST /user': 'UserController.create'` should be outside of the object...

**Review by aichbauer — CHANGES_REQUESTED** [review]

**Comment by aichbauer** [comment]
We should also test it with `import` in `test/src/index.js`

**Comment by toksdotdev** [comment]
Ok, I'll work on the changes.

**Review by alkanyunus — APPROVED** [review]

**Review by alkanyunus — COMMENTED** [review]
I'm taking my approvement back :)

File: src/index.js (hunk 2 above)
Comment (toksdotdev) [reply src/index.js:82]: Yes, it will directly call the controller method.

File: test/src/require.js (hunk 1 above)
Comment (toksdotdev) [reply test/src/require.js]: Oops.

**Comment by toksdotdev** [comment]
Just adjusted the PR based on the previous reviews (testing with imports included).

Also contained is @joneldiablo PR to allow grouped middlewares on a route path.
```Javascript
const groupedMiddleware1 = (req, res, next) => {
  next();
};

const router = mapRoutes(routes, 'test/fixtures/controllers/', [groupedMiddleware1]);
```

**Comment by aichbauer** [comment]
@TNkemdilim right now it is also possible to pass a function as third argument like:

```js
const router = mapRoutes(routes, 'test/fixtures/controllers/', middlewareGeneral);
```

If we support this, we should also have a test case for this case...

**Comment by aichbauer** [comment]
@TNkemdilim any updates for this test case? If we write that last test i am happy to merge this PR. Also thank you for taking the time and contribute to this project, i really appreciate your effort 👍

**Comment by toksdotdev** [comment]
I'll send in the updates shortly.

**Comment by toksdotdev** [comment]
Done!!! Fixes #20  🚀🚀🚀

**Review by aichbauer — APPROVED** [review]

**Pull request merged**